src/ServiceClient/ISocketManager.cs

```diff
@@ -0,0 +1,9 @@
+using System.Net.WebSockets;
+
+namespace DeribitServiceClient.ServiceClient;
+
+public interface ISocketManager
+{
+    ClientWebSocket GetSocket();
+    Task ConnectAsync(ClientWebSocket socket);
```

Comment:
I suggest this should be something like Task<ClientWebSocket?> ConnectAsync(ClientWebSocket socket).
The ConnectAsync method should return whether it could be connected or not. The advantages of having return an instance of the ClientWebSocket is that the console could use it.
Also, in line 7, could it be ClientWebSocket? GetSocket()? what happen if there is not a socket?